internal/services/compute/encryption_settings.go

```diff
@@ -63,10 +63,15 @@ func encryptionSettingsSchema() *pluginsdk.Schema {
 	}
 }
 
-func expandManagedDiskEncryptionSettings(settings map[string]interface{}) *compute.EncryptionSettingsCollection {
-	enabled := settings["enabled"].(bool)
+func expandManagedDiskEncryptionSettings(settingsList []interface{}) *compute.EncryptionSettingsCollection {
+	if len(settingsList) == 0 {
+		return &compute.EncryptionSettingsCollection{}
```

Comment:
This is for update call, when removing the `encryption_settings` block. This will set the enabled to false and remove encryption keys